hackathon_site/dashboard/frontend/src/pages/Inventory/Inventory.test.tsx

```diff
@@ -34,13 +35,48 @@ describe("Inventory Page", () => {
             .calledWith(categoriesUri)
             .mockResolvedValue(categoryApiResponse);
 
-        const { getByText } = render(<Inventory />);
+        const { getByText, queryByText } = render(<Inventory />);
 
         await waitFor(() => {
             expect(getByText(mockHardware[0].name)).toBeInTheDocument();
             expect(getByText(mockCategories[0].name)).toBeInTheDocument();
         });
 
+        // by default the hardwareUri has a limit of 100
+        if (mockHardware.length <= 100) {
+            expect(queryByText(/load more/i)).not.toBeInTheDocument();
+        } else {
+            expect(getByText(/load more/i)).toBeInTheDocument();
+        }
+    });
+
+    test("Hardware count", async () => {
```

Comment:
I'm not sure how I should test the hardware count when filtering by category, I was thinking returning a filtered response from the api but that feels like cheating lol. The other way is going through the process of applying the filter and then checking but we already test the filtering part in another test so that feels redundant. What do you guys think?

Reply:
Is that something we need to test?

We already know that:
- The hardware API will always return a `count`. This is guaranteed by our backend
- That filtering hits the same hardware API, just with filter parameters applied. The inventory filter tests guarantee this.
    - Side note - looks like I never wrote a test for the actual api `get` function, so no guarantees that it's actually building filter params correct. Whoops 😬 . If you wanted to do that, you could create an api.test.ts file and use jest to mock axios (or just make a backlog ticket for it)

So in short, applying filters shouldn't have any effect on the `count` parameter, so there's no need to write tests that include applying filters (unless I'm missing something).

Reply:
You should break this into two test cases - one where the load more button appears, and one where it doesn't.

Something like

```typescript
it("Shows count and load more button when there is more hardware to fetch", async () => {
    ...
});

it("Shows count and no load more button when there is no more hardware to fetch", async () => {
    ...
});
```

Descriptive test names are also preferred (which is why I prefer using `it()` since it reads nicer).

Reply:
Ah okay, I used test() just to see what you think lol
The reason why I didn't break it up was because I thought the first test case would not show the load more button (since I don't think we'd mock 100 items) and so if I made another test for it, the test would look almost the same as the first test.

Reply:
Ignore the comment above, I read your previous comments (sometimes github shows the comments out of order)